### PR TITLE
Cellranger 10XV4

### PR DESCRIPTION
**Pull request by nick-youngblut**
Added support for `10XV4` (`SC3Pv4`). This includes:
*  updating `protocols.json`
* adding the 10X V4 barcode whitelist
* updating the docs

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)
  - `nf-core schema build` fails due to:  `--fasta: Not in pipeline parameters` & `--gtf: Not in pipeline parameters.`
- [X] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [X] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/kallistobustools/count/meta.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ input:
   - technology:
       type: string
       description: String value defining the sequencing technology used.
-      pattern: "{10XV1,10XV2,10XV3,CELSEQ,CELSEQ2,DROPSEQ,INDROPSV1,INDROPSV2,INDROPSV3,SCRUBSEQ,SURECELL,SMARTSEQ}"
+      pattern: "{10XV1,10XV2,10XV3,10XV4,CELSEQ,CELSEQ2,DROPSEQ,INDROPSV1,INDROPSV2,INDROPSV3,SCRUBSEQ,SURECELL,SMARTSEQ}"
```
Comment (grst) [inline modules/nf-core/kallistobustools/count/meta.yml:50]: This change would need to be made in `nf-core/modules` and then be pulled into the pipeline using `nf-core modules update`.

**Review by grst — APPROVED** [review]
Many thanks for working on this! Looks mostly good to me, could you please
 * update the CHANGELOG 
 * reset the kallistobustools module (nf-core doesn't allow modification of modules installed from nf-core/modules, they have to be updated centrally in the modules repository). You can follow up on that separately - the pattern is only documentation and has no effect on the pipeline functionality.

**Comment by nick-youngblut** [comment]
@grst for "update the CHANGELOG", is this sufficient:

```
## v2.7.1 - 2024-07-29

- Add 10XV4 chemistry
```

I can also bump the version to 2.7.1 throughout the codebase: `nextflow.config` & `multiqc_config.yml`

**Comment by grst** [comment]
Don't worry about the version, I'll do that when making a patch release

**Comment by grst** [comment]
@nf-core-bot fix linting

**Review by grst — APPROVED** [review]

**Pull request merged**